kv/memberlist/memberlist_client.go

```diff
@@ -584,15 +591,18 @@ func (m *KV) joinMembersOnStartup(ctx context.Context) bool {
 		// This is harmless and simpler.
 		nodes := m.discoverMembers(ctx, m.cfg.JoinMembers)
 
-		reached, err := m.memberlist.Join(nodes) // err is only returned if reached==0.
-		if err == nil {
-			level.Info(m.logger).Log("msg", "joining memberlist cluster succeeded", "reached_nodes", reached, "elapsed_time", time.Since(startTime))
-			return true
+		if len(nodes) > 0 {
+			reached, err := m.memberlist.Join(nodes) // err is only returned if reached==0.
+			if err == nil {
+				level.Info(m.logger).Log("msg", "joining memberlist cluster succeeded", "reached_nodes", reached, "elapsed_time", time.Since(startTime))
+				return true
+			}
+			level.Warn(m.logger).Log("msg", "joining memberlist cluster: failed to reach any nodes", "retries", boff.NumRetries(), "err", err)
+			lastErr = err
+		} else {
+			level.Warn(m.logger).Log("msg", "joining memberlist cluster: failed to resolve nodes", "retries", boff.NumRetries())
```

Comment:
```suggestion
			level.Warn(m.logger).Log("msg", "joining memberlist cluster: found no nodes to join", "retries", boff.NumRetries())
```

We don't know if it failed, or just nothing was found.

Reply:
ack